src/lib/geogram/voronoi/RVD.cpp

```diff
@@ -1295,23 +1364,44 @@ namespace {
             ) {
                 geo_argused(v_adj);
                 geo_argused(t_adj);
-                f_ += simplex_func_->eval(
+                auto res = simplex_func_->eval(
                     v,v1,v2,v3,t,index_t(t_adj),index_t(v_adj)
                 );
+                if (multithread_) {
+                    accu_f_.push_back(res);
+                }
+                else {
+                    f_ += res;
+                }
             }
 
             /**
-             * \brief Gets the function value.
+             * \brief Gets the function value.  Once called, this object ceases
+             *  to be multithread-safe (though simultaneously using the 
+             *  underlying IntegrationSimplex from multiple instances of this 
+             *  class remains ok).
              * \return The function value accumulated so far.
              */
-            double f() const {
+            double f() {
+                if (multithread_) {
```

Comment:
If you call this from two threads simultaneously, yes.  Which you shouldn't do, since once you call the first one it ceases to be multithread-safe and calling from the second thread simultaneously is using a non-multithread-safe object from two threads simultaneously.